mocks/moderc3156/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+#![no_std]
+
+use soroban_sdk::{
+    contract, contractimpl, contracttype, Address, Env
+};
+
+#[contracttype]
+pub enum DataKey {
+    Admin
+}
+
+#[contract]
+pub struct FlashLoanReceiverModifiedERC3156;
+
+#[contractimpl]
+impl FlashLoanReceiverModifiedERC3156 {
+    pub fn init(env: Env, admin: Address) {
+        admin.require_auth();
+        env.storage().instance().set(&DataKey::Admin, &admin);
+    }
+
+    pub fn exec_op(env: Env, caller: Address, _token: Address, _amount: i128, _fee: i128) {
+        // require auth for the flash loan
+        caller.require_auth(); // if you want to allow exec_op to be initiated by only a pool you can do so here.
+        env.storage().instance().get::<DataKey, Address>(&DataKey::Admin).unwrap().require_auth();
+
+        // perform operations here
+        // ...
+
```

Comment:
should this test contract just return `amount` to `caller`?

That way, we can test more common cases of flash loans like:

1. Flash loan 1000 USDC to contract that does arb are returns 1000 USDC
2. Repay 1000 USDC

And that the tokens are being properly transferred to the flash loan contract by the pool